scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/TestPersistentAnchor.java

```diff
@@ -0,0 +1,587 @@
+
+  @Test
+  public void testCaseStartAtStartAnchorAt11() throws SchedulingInterruptedException{
+    TestData testData = createTestCaseStartsAt(PersistentTimeAnchor.START, true, true, 2, 0, 20, TimeAnchor.END, null, 0);
+    assertTrue(testData.plan.isPresent());
+    assertEquals(9, testData.plan.get().getActivitiesById().size());
+    assertTrue(allAnchorsIncluded(testData));
+    assertTrue(checkAnchoredActivities(testData, true, true));
+    assertTrue(checkUnanchoredActivities(testData));
+  }
+
+  //Anchor at END
+  @Test
+  public void testCaseStartAtEndAnchorAt00() throws SchedulingInterruptedException{
+    TestData testData = createTestCaseStartsAt(PersistentTimeAnchor.END, false, false, 2, 0, 20, TimeAnchor.END, null, 0);
+    assertTrue(testData.plan.isPresent());
+    assertEquals(6, testData.plan.get().getActivitiesById().size());
+    assertTrue(allAnchorsIncluded(testData));
+    assertTrue(checkAnchoredActivities(testData, true, false));
+    assertTrue(checkUnanchoredActivities(testData));
+  }
+
+  @Test
+  public void testCaseStartAtEndAnchorAt01() throws SchedulingInterruptedException{
+    TestData testData = createTestCaseStartsAt(PersistentTimeAnchor.END, false, true, 2, 0, 20, TimeAnchor.END, null, 0);
+    assertTrue(testData.plan.isPresent());
+    assertEquals(6, testData.plan.get().getActivitiesById().size());
+    assertTrue(allAnchorsIncluded(testData));
+    assertTrue(checkAnchoredActivities(testData, true, false));
+    assertTrue(checkUnanchoredActivities(testData));
+  }
+
+  @Test
+  public void testCaseStartAtEndAnchorAt10() throws SchedulingInterruptedException{
+    TestData testData = createTestCaseStartsAt(PersistentTimeAnchor.END, true, false, 2, 0, 20, TimeAnchor.END, null, 0);
+    assertTrue(testData.plan.isPresent());
+    assertEquals(6, testData.plan.get().getActivitiesById().size());
+    assertTrue(allAnchorsIncluded(testData));
+    assertTrue(checkAnchoredActivities(testData, true, true));
+    assertTrue(checkUnanchoredActivities(testData));
+  }
+
+  @Test
+  public void testCaseStartAtEndAnchorAt11() throws SchedulingInterruptedException{
+    TestData testData = createTestCaseStartsAt(PersistentTimeAnchor.END, true, true, 2, 0, 20, TimeAnchor.END, null, 0);
+    assertTrue(testData.plan.isPresent());
+    assertEquals(9, testData.plan.get().getActivitiesById().size());
+    assertTrue(allAnchorsIncluded(testData));
+    assertTrue(checkAnchoredActivities(testData, true, true));
+    assertTrue(checkUnanchoredActivities(testData));
+  }
+
+
+  /* Test cases in which the goal is created with TimeExpression endsAt
+   */
+
+  public TestData createTestCaseStartsAt(PersistentTimeAnchor persistentAnchor,  boolean missingActAssociationsWithAnchor, boolean missingActAssociationsWithoutAnchor, int activityDurationHours, int goalStartPeriodHours, int goalEndPeriodHours, TimeAnchor timeAnchor, TimeExpressionRelative timeExpression, long relativeOffsetHours)
+  throws SchedulingInterruptedException
```

Comment:
all of these parameters should be `final`. Please do a pass over this file for final modifier, I saw other occurrences.

Reply:
all TestPersistentAnchor reviewed/updated with (final) var when applicable